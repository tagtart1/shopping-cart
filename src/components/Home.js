import "../styles/Home.css";
import { Link } from "react-router-dom";
import { useEffect } from "react";

function Home() {
  useEffect(() => {
    document.body.style.removeProperty("background");
    const navLinks = document.querySelectorAll(".link-item");
    const cartIcon = document.querySelector(".cart-icon");
    const cartQuantity = document.querySelector(".quantity-cart-display");
    navLinks.forEach((link) => {
      link.style.color = "white";
      link.classList.remove("catalog-underline");
    });
    if (cartIcon) cartIcon.classList.remove("icon-catalog");
    if (cartQuantity) {
      cartQuantity.classList.remove("cart-quantity-catalog");
      cartQuantity.classList.add("cart-quantity-home");
    }
  }, []);

  return (
    <div className="home">
      <p className="small-main-text">ONE OF THE FROG STORES OF ALL TIME</p>
      <p className="big-main-text">
        Get the best frogs, the frog juice, and more frogs
      </p>
      <Link to={"/catalog"}>
        <button type="button" className="home-shop-now-btn">
          Shop Now
        </button>
      </Link>
    </div>
  );
}

export default Home;
