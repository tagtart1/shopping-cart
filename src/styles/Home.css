* {
  box-sizing: border-box;
}

body {
  background-image: url("../images/frog-background.jpg");
}

.home {
  display: flex;
  justify-content: center;
  flex-direction: column;
  color: white;
  height: auto;
  flex-grow: 1;
  margin-left: 100px;
  margin-bottom: 200px;
  width: 500px;
}
.home-shop-now-btn {
  font-size: 48px;
  width: 500px;
  padding: 24px;
  border-radius: 48px;
  font-weight: bolder;
  cursor: pointer;
}

.small-main-text {
  color: rgb(221, 208, 90);
  font-size: 20px;
  font-weight: bold;
  letter-spacing: 2px;
}

.big-main-text {
  font-size: 48px;
  margin-bottom: 48px;
  font-weight: bold;
}

p {
  margin: 0;
}
