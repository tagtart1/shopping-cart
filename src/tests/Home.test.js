import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import Home from "../components/Home";

describe("Home component", () => {
  it("Renders page correctly", () => {
    const { container } = render(<Home />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
});
