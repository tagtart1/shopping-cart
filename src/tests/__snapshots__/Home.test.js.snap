// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders page correctly 1`] = `
<div>
  <div
    class="home"
  >
    <p
      class="small-main-text"
    >
      ONE OF THE FROG STORES OF ALL TIME
    </p>
    <p
      class="big-main-text"
    >
      Get the best frogs, the frog juice, and more frogs
    </p>
    <a
      href="/catalog"
    >
      <button
        class="home-shop-now-btn"
        type="button"
      >
        Shop Now
      </button>
    </a>
  </div>
</div>
`;
